From 23df384bc373985b8faa559189c38a53db5f00f7 Mon Sep 17 00:00:00 2001
From: Marcel Mulder <mmulder@victronenergy.com>
Date: Tue, 18 Mar 2014 16:43:19 +0100
Subject: [PATCH 1/5] Add set hwclock when time is decoded

---
 src/ntp.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ntp.c b/src/ntp.c
index dd246eb9cba9..01f5dc605a22 100644
--- a/src/ntp.c
+++ b/src/ntp.c
@@ -367,6 +367,9 @@ static void decode_msg(void *base, size_t len, struct timeval *tv,
 			connman_error("Failed to set time");
 			return;
 		}
+		if(system("hwclock --utc --systohc")) {
+			connman_error("Failed to set hwclock");
+		}
 
 	}
 }
-- 
2.10.0

