From 817c9a486b01b5f2da79d1040dbc0a482d0298ce Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Sun, 19 Jun 2016 20:37:01 +0100
Subject: [PATCH] bpp3: update default environment for swupdate

---
 include/configs/bpp3.h | 43 ++++++++++++++++++++++++++++++-------------
 1 file changed, 30 insertions(+), 13 deletions(-)

diff --git a/include/configs/bpp3.h b/include/configs/bpp3.h
index 771e783cc766..f1a15470abb3 100644
--- a/include/configs/bpp3.h
+++ b/include/configs/bpp3.h
@@ -82,27 +82,44 @@
 # define CONFIG_PREBOOT "usb start"
 #endif
 
-#ifdef MTDPARTS_DEFAULT
-# undef MTDPARTS_DEFAULT
-#endif
-#define MTDPARTS_DEFAULT	"mtdparts=omap2-nand.0:512k(MLO),"\
-		"1m(u-boot),256k(env1),256k(env2),1m(u-boot2),256k(bootparms),"\
-		"768k(splash),6m(kernel),200m(data),-(rootfs)"
+#undef MTDPARTS_DEFAULT
+#define MTDPARTS_DEFAULT "mtdparts=omap2-nand.0:"			\
+	"128k(spl1),128k(spl2),128k(spl3),128k(spl4),"			\
+	"1m(u-boot),256k(env1),256k(env2),1m(u-boot2),"			\
+	"256k(bootparms),768k(splash),6m(kernel1),6m(kernel2),-(ubisystem)"
 
 #define	CONFIG_EXTRA_ENV_SETTINGS \
 	"mmcdev=0\0" \
 	"loadbootscript=fatload mmc ${mmcdev} 80004000 boot.scr\0" \
 	"bootcmd=" \
-		"if mmc rescan ${mmcdev}; then " \
-			"echo SD/MMC found on device ${mmcdev};" \
-			"if run loadbootscript; then " \
-				"echo Running bootscript from mmc ...;" \
+		"if gpio input 29; then " \
+			"echo Checking USB;" \
+			"usb start;" \
+			"if fatload usb 0 80004000 boot.scr; then " \
+				"echo Starting from USB;" \
+				"setenv boot_dev usb;" \
 				"source 80004000;" \
 			"fi;" \
+			"if mmc rescan ${mmcdev}; then " \
+				"echo SD/MMC found on device ${mmcdev};" \
+				"if run loadbootscript; then " \
+					"echo Starting from mmc;" \
+					"setenv boot_dev mmc;" \
+					"source 80004000;" \
+				"fi;" \
+			"fi;" \
 		"fi;" \
-		"run nandboot;\0" \
-		"loadaddr=80300000\0" \
-		"nandboot=nand read ${loadaddr} kernel && bootm ${loadaddr}\0"
+		"setenv boot_dev nand;" \
+		"setenv kernel kernel${version};" \
+		"setenv rootfs ubi0:rootfs${version};" \
+		"setenv bootargs root=${rootfs} rootfstype=ubifs ubi.mtd=12 " \
+			"${miscargs} ${mtdparts};" \
+		"run nandboot\0" \
+	"version=1\0" \
+	"loadaddr=80300000\0" \
+	"miscargs=console=ttyO0,115200 omapdss.def_disp=lcd omapfb.vram=0:2M vram=2M fbskip\0" \
+	"mtdparts=" MTDPARTS_DEFAULT "\0" \
+	"nandboot=nand read ${loadaddr} ${kernel} && bootm ${loadaddr}\0"
 
 /* SPL OS boot options */
 #define CONFIG_CMD_SPL
-- 
2.9.0

